### ISSUE TITLE
Mask-RCNN (pretrained on COCO) on higher resolution images (3MP)

### ISSUE THREAD

**Issue by habjoel** [open-mmlab/mmdetection#8409]
## Problem statement
Hi there, 

I have been using the following [Mask-RCNN model](https://github.com/open-mmlab/mmdetection/blob/ca11860f4f3c3ca2ce8340e2686eeaec05b29111/configs/mask_rcnn/mask_rcnn_r50_fpn_1x_coco.py). I took the pretrained COCO weights and did some transfer learning on it to detect 7 different classes of waste using the [TACO Dataset](http://tacodataset.org/) and some own images. 

Under [/configs/_base_/datasets/coco_instance.py](https://github.com/open-mmlab/mmdetection/blob/ca11860f4f3c3ca2ce8340e2686eeaec05b29111/configs/_base_/datasets/coco_instance.py#L20), I have seen that every image is rescaled to a range of (1333, 800) for training and inference. 

However, now I would like to use this model to do  inference on 3MP images (1544x2064). 
I tried to first just test the model in inference on some 3MP images. Therefore, I changed [this line](https://github.com/open-mmlab/mmdetection/blob/ca11860f4f3c3ca2ce8340e2686eeaec05b29111/configs/_base_/datasets/coco_instance.py#L20) to `img_scale=(2064, 1544)`. 

The outcome was really bad. So I was advised in [this issue](https://github.com/open-mmlab/mmdetection/issues/8175) to change the [anchor scale](https://github.com/open-mmlab/mmdetection/blob/ca11860f4f3c3ca2ce8340e2686eeaec05b29111/configs/_base_/models/mask_rcnn_r50_fpn.py#L25) from `scales=[8]` to `scales=[12]` or `scales=[16]` but there was not much improvement. 
Do I also need to change `ratios` and `strides` somehow?

I am starting to wonder if I need to train the model newly from scratch if I want to use a higher resolution than (1333, 800) (train with COCO from scratch and scale all the images to a input size of (2064, 1544)). Or is there an easier way to achieve my goal of accurate predictions at 3MP? 

Of course, I would like to make use of COCO pretraining since it's a very large dataset and I would prefer not having to train it all again from scratch.

I have asked about this problem already on [this issue](https://github.com/open-mmlab/mmdetection/issues/8175) and on [this issue on MMDeploy](https://github.com/open-mmlab/mmdeploy/issues/552) but I thought I will open the discussion in a new issue for the benefit of the community.

Any help and insight on this is very much appreciated!  

### Raw image (clearly a plastic bottle)
![frame000700_1](https://user-images.githubusercontent.com/96238916/180414540-ac6a8755-0416-4760-ba1d-bc279c2bc1f2.jpg)

### Predictions rescaled to (1333, 800) --> very good prediction
![pred_frame000700_1](https://user-images.githubusercontent.com/96238916/180414880-852b2aee-0eeb-4ba6-9994-f0800c99e68a.jpg)

### Predictions rescaled to (2064, 1544) --> bad prediction
![pred_res_frame000700_1](https://user-images.githubusercontent.com/96238916/180415392-69f38597-f669-4f40-9038-3f1082ecf063.jpg)

### Raw image (2 plastic bottles and one helmet which should be classified as "other")
![frame001200_1](https://user-images.githubusercontent.com/96238916/180416227-5acf1134-0c87-4414-b0df-fac2cdbff213.jpg)

### Predictions rescaled to (1333, 800) --> very good prediction
![pred_frame001200_1](https://user-images.githubusercontent.com/96238916/180415862-a3c3659b-1336-4956-a7ed-dba6e5f6c078.jpg)

### Predictions rescaled to (2064, 1544 --> bad prediction
![pred_res_frame001200_1](https://user-images.githubusercontent.com/96238916/180416655-f7991945-503b-4ec6-a815-a9feff740251.jpg)

I could go on but I think the issue should be more or less clear.

**Comment by BIGWangYuDong** [comment]
I think maybe re-train a new model with higher resolution shape can achieve higher performance, because the features may not match the higher resolution

**Comment by habjoel** [comment]
thank you @BIGWangYuDong.

I have been trying to retrain the COCO pretrained model using the [TACO](http://tacodataset.org/) dataset and some own images by setting img_scale to `img_scale=(2064, 1544)` and the anchor scale to `scales=[12]`. It has not brought much better results.
Below, I provide the plots of the training losses during retraining. I do not really understand the difference between `loss_rpn_bbox` vs. `loss_bbox` and `loss_rpn_cls` vs. `loss_cls`. Maybe you can tell me?

Also, based on these plots... could you give any suggestion on what I should do differently? I see that `loss_rpn_bbox` and in `loss_rpn_cls`  rise a lot in the beginning. Is the reason for this that I am using a different anchor size? Maybe I did not train for enough epochs. However, I tested the model with the checkpoint from epoch 9 and the checkpoint from epoch 12 and there was not really an improvement.

I am trying to solve the problem without having to train with COCO from scratch since COCO has 330K images and it would take for ever.

Any suggestion is welcome!

## Below you can see the losses plotted over the course of training:

### loss_rpn_bbox
![image](https://user-images.githubusercontent.com/96238916/182366693-ea44975c-201f-4b70-95d9-d60f1eb5ff0d.png)

### loss_rpn_cls
![image](https://user-images.githubusercontent.com/96238916/182367103-95c0a7b2-6eae-4fdb-9518-c2a1cf65f25f.png)

### loss_cls, loss_bbox, loss_mask ant the total loss together in one plot
![image](https://user-images.githubusercontent.com/96238916/182367240-de57c6c7-055d-49ff-8657-a35fa3638cb7.png)
